app/aboot/fastboot-extra.c

```diff
@@ -13,6 +14,87 @@
 #include <target.h>
 #include "fastboot.h"
 
+static void cmd_oem_readl(const char *arg, void *data, unsigned sz)
+{
+	char response[MAX_RSP_SIZE];
+	uint32_t addr = 0;
+
+	addr = atoi(arg);
+	snprintf(response, sizeof(response), "0x%08x\n", readl(addr));
+	fastboot_info(response);
+
+	fastboot_okay("");
+}
+
+static void cmd_oem_writel(const char *arg, void *data, unsigned sz)
+{
+	char response[MAX_RSP_SIZE], address[11] = {0};
+	uint32_t addr = 0, value = 0;
+	int i;
+
+	for (i = 0; ; i++) {
+		if (!arg[i]) {
+			fastboot_fail("");
+			return;
+		}
+		if (arg[i] == ' ' && i <= 10 && address[0] == 0) {
+			strncpy(address, arg, i);
+			addr = atoi(address);
+		}
+		if (arg[i] == ' ' && arg[i + 1] != ' ') {
+			value = atoi(arg + i + 1);
+			break;
+		}
+	}
```

Comment:
Looks like we can share this code between writel and writeb. I'm also a bit confused about how it works... Perhaps we should simply use the standard `strtok` here? Something like this:

```c
static bool parse_write_args(const char *arg, uint32_t *addr, uint32_t *value)
{
	char *saveptr;
	char *args = strdup(arg);
	char *addr_str = strtok_r(args, " ", &saveptr);
	char *value_str = strtok_r(NULL, " ", &saveptr);

	if (!addr_str || !value_str) {
		free(args);
		return false;
	}

	*addr = atoi(addr_str);
	*value = atoi(value_str);
	free(args);
	return true;
}
```

Reply:
> we can share this code between writel and writeb. 

Yeah, I thought about that too. 

>we should simply use the standard strtok here

Ah, I thought that won't work with more than one space. Turned out I thought too much.
Thanks for the review!